### ISSUE TITLE
Allow searching for partial word matches

### ISSUE THREAD

**Issue by sagamusix** [demozoo/demozoo#137]
Right now, it's not possible to search for partial words, e.g. "vivi" doesn't bring up any results but "vivid" does. It would be nice if partial word matches would also be possible, as on pouet. It can be helpful when you don't remember the exact spelling or full name of a demo (or if you're lazy...)

**Comment by asbjornu** [comment]
What sort of index is backing the search right now? Would partial searches be easy to implement in the existing index or would setting up something like Elasticsearch or similar be a viable or perhaps even required solution?

**Comment by gasman** [comment]
It's using Postgres full-text search, which is comparable in functionality to Elasticsearch. (The automatic suggestions-as-you-type are a separate mechanism, powered by a simple substring match.)

FWIW, I don't think this behaviour is desirable, at least by default: if you search for 'rob', you're looking for the scener called Rob, not an assortment of things that have the substring 'rob' in them. I'd be OK with supporting partial word matches with wildcards (e.g. `rob*`), but I don't think "proper" search engines are really designed for that (since they index the text on a per-token basis).

**Comment by sagamusix** [comment]
I agree to the power of wildcards, but completely disregarding that - I guess any full matches should rank above partial matches (i.e. Rob should always be above Robin in that scenario).

**Comment by asbjornu** [comment]
I'm not very familiar with Postgres' full-text search, but it seems like it supports distance searches, so a search for `rob` would match the name `rob` with a higher score than `robin` and `robert`, thus [ranking by distance](https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-RANKING) would put `rob` at the top with a score of `1`.

Any match with a score of `1` could be given special prominence in the search results compared matches with a lower score. This might also make it easier to fix #91, since all metadata could be placed in the same document and then matching words could just be [highlighted](https://www.postgresql.org/docs/current/textsearch-controls.html#TEXTSEARCH-HEADLINE). Just a thought.
